geyser-plugin-manager/src/geyser_plugin_manager.rs

```diff
@@ -4,12 +4,48 @@ use {
     libloading::Library,
     log::*,
     solana_geyser_plugin_interface::geyser_plugin_interface::GeyserPlugin,
-    std::path::Path,
+    std::{
+        ops::{Deref, DerefMut},
+        path::Path,
+    },
 };
 
+#[derive(Debug)]
+pub struct LoadedGeyserPlugin {
+    name: String,
```

Comment:
Maybe libname? I do not see how this utilized to solve the problem. We intentionally request the plugin has different names so that load/unload can work reliably. Why cannot you give different names to different plugins?

Reply:
> Why cannot you give different names to different plugins?

because for a new name you need to re-compile the whole plugin with hardcoded name in the code :slightly_smiling_face: https://github.com/solana-labs/solana/blob/666ce9b3befbf5e8bab7bc827d375d018a96adad/geyser-plugin-interface/src/geyser_plugin_interface.rs#L290

Reply:
I think this problem can be solved by the plugin itself without the framework's involvement. The &static reference makes it a little tricky but not insurmountable. Here is some sample code to the effect:

{code}
use std::fs;
use std::sync::{Mutex, Once};

use std::collections::HashMap;

struct Plugin {
    name: Mutex<HashMap<i32, &'static str>>,
    initialized: Once,
}

impl Plugin {
    pub fn new() -> Self {
        Self {
            name: Mutex::new(HashMap::default()),
            initialized: Once::new(),
        }
    }

    pub fn load(&mut self, file: String) {
        self.initialized.call_once(|| {
            let file_content: String = fs::read_to_string(file).unwrap();
            let s = Box::new(file_content);
            let s = Box::leak(s);
            self.name.lock().unwrap().insert(1, s);
        })
    }

    fn get_name(&self) -> &'static str {
        let data = self.name.lock().unwrap();
        let data = data.get(&1).unwrap();
        data
    }
}

fn main() {
    let mut s = Plugin::new();
    s.load("config.txt".to_string());
    println!("This is my content {}", s.get_name());
}
{code}

Reply:
haha, nice workaround
I think `name()` happens before `load()` so `call_once` should be moved in this case

while this code probably would work I still think that supporting `libname` through config would be nice